nats.go

```diff
@@ -2563,6 +2577,13 @@ func (nc *Conn) connectProto() (string, error) {
 		pass = o.Password
 		token = o.Token
 		nkey = o.Nkey
+
+		if nc.Opts.UserInfo != nil {
```

Comment:
...this tiny change will have huge opportunities for our organization (from #1694)!